### DISCUSSION TITLE
Initial motor positions

### DISCUSSION THREAD

**Discussion by perceptualrobots** [cyberbotics/wrestling#70] [Q&A] [answered]
Hi,

I am assuming that if I want to move a motor to a particular position I set a value with setPosition(...) and the motor will move until it reaches that target position. Is that correct?

First, I am trying to read the initial positions of the Nao motors, though I don't seem to be getting the correct values.

Looking at the initial posture of the robot and https://www.cyberbotics.com/doc/guide/nao, I conclude that the values should all be zero, is that right?

However, when I read the initial values, with getTargetPosition(), I get one of the following:

```
initial: {'LHipPitch': -1.774, 'LKneePitch': 2.113, 'LAnklePitch': -1.189, 'RHipPitch': 0.484, 'RKneePitch': -0.092, 'RAnklePitch': 0.932}

initial: {'LHipPitch': 0.484, 'LKneePitch': 2.113, 'LAnklePitch': 0.923, 'RHipPitch': 0.484, 'RKneePitch': 2.113, 'RAnklePitch': 0.932}
```

These seem to be the min/max values of the motors, rather than the initial values, why am I getting those?

Any idea what I am doing wrong?

Here is the order of main code lines:

```
    wrestler = WrestlerSupervisorServer()
    while True:
        wrestler.simulationReset()
        wrestler.run()
```

 ```
def run(self):
        self.initSupervisor()
        self.motion_library = MotionLibrary()
        ...
        initial = readLegs()
        while loop
```

```
    def readLegs(self):
        
        lhp = self.LHipPitch.getTargetPosition()  
        ...      
        
        legs = {'LHipPitch': round(lhp, 3), 'LKneePitch': round(lkp,3), 'LAnklePitch': round(lap, 3), 'RHipPitch': round(rhp, 3), 'RKneePitch': round(rkp, 3), 'RAnklePitch': round(rap, 3)}
        
        return legs
```

```
    def initSupervisor(self):
        self.robot = [0] * 2
        self.robot[1] = self.getFromDef('WRESTLER_BLUE').getFromProtoDef('HEAD_SLOT')
        self.robot[0] = self.getFromDef('WRESTLER_RED').getFromProtoDef('HEAD_SLOT')
        self.min = [[0] * 3 for i in range(2)]
        self.max = [[0] * 3 for i in range(2)]
        for i in range(2):
            self.min[i] = self.robot[i].getPosition()
            self.max[i] = self.robot[i].getPosition()
        self.coverage = [0] * 2
        self.ko_count = [0] * 2

```

**Comment by omichel** [comment, marked as answer]
You should rely on the [position sensors](https://cyberbotics.com/doc/reference/positionsensor) to get the position of the motors. These devices are named "HeadPitchS", "HeadYawS", "LAnklePitchS", etc. on the Nao robot. `getTargetPosition()` simply returns the last target position of the motor that you defined, but it doesn't necessarily correspond to the actual position reached by the motor.

**Reply by perceptualrobots** [reply]
Thanks. 

So, to read the motor position I do this,

```
self.LAnklePitchS = robot.getDevice("LAnklePitchS")
sensor_reading = self.LAnklePitchS.getValue()

```
To get the motor to move to a postion I want I do this,

```
self.LAnklePitchM = robot.getDevice("LAnklePitch")
self.LAnklePitchM.setPosition(desired_position)

```
Is that correct?

**Reply by omichel** [reply]
Yes, that's correct.
